### ISSUE TITLE
manual/scalaGuide/main/http/ScalaResults.md

### ISSUE THREAD

**Issue by garbagetown** [playframework-ja/translation-project#612]
- https://github.com/garbagetown/playdocja/blob/2.2.0/documentation/2.2.0/manual/scalaGuide/main/http/ScalaResults.md

```
--- /Users/garbagetown/Desktop/2.1.5/manual/scalaGuide/main/http/ScalaResults.md    2013-09-19 22:53:02.000000000 +0900
+++ /Users/garbagetown/Desktop/2.2.0/manual/scalaGuide/main/http/ScalaResults.md    2013-09-19 10:11:22.000000000 +0900
@@ -6,31 +6,24 @@

 For example:

-'''scala
-val textResult = Ok("Hello World!")
-'''
+@[content-type_text](code/ScalaResults.scala)
+

 Will automatically set the `Content-Type` header to `text/plain`, while:

-'''scala
-val xmlResult = Ok(<message>Hello World!</message>)
-'''
+@[content-type_xml](code/ScalaResults.scala)

-will set the Content-Type header to `text/xml`.
+will set the Content-Type header to `application/xml`.

 > **Tip:** this is done via the `play.api.http.ContentTypeOf` type class.

 This is pretty useful, but sometimes you want to change it. Just use the `as(newContentType)` method on a result to create a new similar result with a different `Content-Type` header:

-'''scala
-val htmlResult = Ok(<h1>Hello World!</h1>).as("text/html")
-'''
+@[content-type_html](code/ScalaResults.scala)

 or even better, using:

-'''scala
-val htmlResult = Ok(<h1>Hello World!</h1>).as(HTML)
-'''
+@[content-type_defined_html](code/ScalaResults.scala)

 > **Note:** The benefit of using `HTML` instead of the `"text/html"` is that the charset will be automatically handled for you and the actual Content-Type header will be set to `text/html; charset=utf-8`. We will see that in a bit.

@@ -38,12 +31,7 @@

 You can also add (or update) any HTTP header to the result:

-'''scala
-Ok("Hello World!").withHeaders(
-  CACHE_CONTROL -> "max-age=3600", 
-  ETAG -> "xx"
-)
-'''
+@[set-headers](code/ScalaResults.scala)

 Note that setting an HTTP header will automatically discard the previous value if it was existing in the original result.

@@ -53,48 +41,28 @@

 You can easily add a Cookie to the HTTP response using:

-'''scala
-Ok("Hello world").withCookies(
-  Cookie("theme", "blue")
-)
-'''
+@[set-cookies](code/ScalaResults.scala)

 Also, to discard a Cookie previously stored on the Web browser:

-'''scala
-Ok("Hello world").discardingCookies("theme")
-'''
+@[discarding-cookies](code/ScalaResults.scala)

 ## Changing the charset for text based HTTP responses.

 For text based HTTP response it is very important to handle the charset correctly. Play handles that for you and uses `utf-8` by default.

-The charset is used to both convert the text response to the corresponding bytes to send over the network socket, and to alterate the `Content-Type` header with the proper `;charset=xxx` extension.
+The charset is used to both convert the text response to the corresponding bytes to send over the network socket, and to update the `Content-Type` header with the proper `;charset=xxx` extension.

 The charset is handled automatically via the `play.api.mvc.Codec` type class. Just import an implicit instance of `play.api.mvc.Codec` in the current scope to change the charset that will be used by all operations:

-'''scala
-object Application extends Controller {
-    
-  implicit val myCustomCharset = Codec.javaSupported("iso-8859-1")
-    
-  def index = Action {
-    Ok(<h1>Hello World!</h1>).as(HTML)
-  }
-    
-}
-'''
+@[full-application-set-myCustomCharset](code/ScalaResults.scala)

 Here, because there is an implicit charset value in the scope, it will be used by both the `Ok(...)` method to convert the XML message into `ISO-8859-1` encoded bytes and to generate the `text/html; charset=iso-8859-1` Content-Type header.

 Now if you are wondering how the `HTML` method works, here it is how it is defined:

-'''scala
-def HTML(implicit codec: Codec) = {
-  "text/html; charset=" + codec.charset
-}
-'''
+@[Source-Code-HTML](code/ScalaResults.scala)

 You can do the same in your API if you need to handle the charset in a generic way.

-> **Next:** [[Session and Flash scopes | ScalaSessionFlash]]
\ No newline at end of file
+> **Next:** [[Session and Flash scopes | ScalaSessionFlash]]

```

**Comment by garbagetown** [comment]
pull reqed at #684

**Comment by wm3** [comment]
merged #684
